client/src/pages/Messages/MessagesDashboard.tsx

```diff
@@ -0,0 +1,250 @@
+  const scrollRef = useRef<HTMLLIElement>(null);
+  const [open, setOpen] = useState(false);
+  const handleOpen = () => setOpen(true);
+  const handleClose = () => setOpen(false);
+
+  const handleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setMessage(event.target.value);
+  };
+
+  useEffect(() => {
+    if (scrollRef.current) {
+      scrollRef.current.scrollIntoView();
+    }
+  }, [chatbox]);
+
+  const handleSubmit = (description: string, conversationId: string) => {
+    setSubmitting(true);
+    sendMessage(description, conversationId).then((data) => {
+      if (data.error) {
+        setSubmitting(false);
+        updateSnackBarMessage(data.error.message);
+      } else if (data.success) {
+        console.log(data.success);
+        updateSnackBarMessage('Sent!');
+        getConversations().then((data) => {
+          if (data.error) {
+            updateSnackBarMessage(data.error.message);
+          } else if (data.success) {
+            const conversations = data.success.conversations;
+
+            setConversations(conversations);
+
+            setSubmitting(false);
+          } else {
+            console.error({ data });
+            setSubmitting(false);
+            updateSnackBarMessage('An unexpected error occurred. Please try again');
+          }
+        });
+        setSubmitting(false);
+      } else {
+        // should not get here from backend but this catch is for an unknown issue
+        console.error({ data });
+
+        setSubmitting(false);
+        updateSnackBarMessage('An unexpected error occurred. Please try again');
+      }
+    });
+  };
+
+  useEffect(() => {
+    setSubmitting(true);
+
+    getConversations().then((data) => {
+      if (data.error) {
+        updateSnackBarMessage(data.error.message);
+      } else if (data.success) {
+        const conversations = data.success.conversations;
+
+        setConversations(conversations);
+
+        setSubmitting(false);
+      } else {
+        console.error({ data });
+        setSubmitting(false);
+        updateSnackBarMessage('An unexpected error occurred. Please try again');
+      }
+    });
+  }, [updateSnackBarMessage]);
+
+  return (
+    <PageContainer>
+      <Box>
+        <Grid container>
+          <Grid item xs={12}>
+            <Typography variant="h6" className="header-message">
+              Inbox Messages
+            </Typography>
+            <Button onClick={handleOpen}>Start new conversation</Button>
+          </Grid>
+        </Grid>
+        <Grid container component={Paper} className={classes.inboxSection}>
+          <Grid item xs={3} className={classes.borderRight500}>
+            <Divider />
+            <List>
+              {conversations.length &&
+                conversations.map((conversation, index) => (
+                  <ListItem button onClick={() => setChatBox(index)} key={conversation._id}>
+                    <ListItemIcon>
+                      <Avatar
```

Comment:
To increase reusability and code maintainability we could probably componentize the sender and receiver chat bubbles